pkg/rekor/identity.go

```diff
@@ -158,6 +149,40 @@ func MatchedIndices(logEntries []models.LogEntry, mvs MonitoredValues) ([]Identi
 					}
 				}
 			}
+
+			oidConstraintSet := make(map[IdentityEntry]struct{})
+			for _, cert := range certs {
+				for _, monitoredCertID := range mvs.CertificateIdentities {
+					certIDMatch, certIDSub, certIDIss, certIDErr := certMatchesPolicy(cert, monitoredCertID.CertSubject, monitoredCertID.Issuers)
+					if certIDErr != nil {
+						return nil, fmt.Errorf("error with policy matching for UUID %s at index %d: %w", uuid, *entry.LogIndex, certIDErr)
+					} else if certIDMatch {
+						matchedEntries = append(matchedEntries, IdentityEntry{
+							CertSubject: certIDSub,
+							Issuer:      certIDIss,
+							Index:       *entry.LogIndex,
+							UUID:        uuid,
+						})
+					}
+				}
+
+				for _, OIDConstraint := range mvs.OIDConstraints {
+					oidMatch, oidErr := oidMatchesPolicy(cert, asn1.ObjectIdentifier(OIDConstraint.OID), OIDConstraint.Constraint)
+					if oidErr != nil {
+						return nil, fmt.Errorf("error with policy matching for UUID %s at index %d: %w", uuid, *entry.LogIndex, oidErr)
+					} else if oidMatch {
+						oidConstraintSet[IdentityEntry{
+							Index: *entry.LogIndex,
+							UUID:  uuid,
+						}] = struct{}{}
+					}
+				}
+			}
+
+			// If any oidConstraints were set, ensure only matched entries that meet all the constraints are returned
+			if len(oidConstraintSet) != 0 {
+				matchedEntries = FilterEntries(matchedEntries, oidConstraintSet)
```

Comment:
I would like the two cert-related sets to be separate - you can monitor by certificate subject alternative name, or you can monitor by OID value. We shouldn't constraint the former based on the latter, otherwise you'd be required to always match by cert subject as well as OID value.

For a concrete example, I may want to monitor for all instances of my repository. I could match against [Source Repo URI](https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264112--source-repository-uri) without the need to match by cert subject.